mmdet/core/general_data/general_data.py

```diff
@@ -112,55 +126,80 @@ def add_meta_info(self, meta):
                         f'img_meta_info {k} has been set as '
                         f'{getattr(self, k)} before, which is immutable ')
             else:
-                self._meta_info_field.add(k)
+                self._meta_info_fields.add(k)
                 self.__dict__[k] = v
 
     def new_results(self):
```

Comment:
Can this function support new_results(data)?

Reply:
Good suggestion